### PR TITLE
Docs: adds a getting started guide for Rancher 2.x

### PR DESCRIPTION
**Pull request by seanmwinn**
Signed-off-by: Sean Winn <sean@isovalent.com>

Adds a getting started guide for Rancher 2.x using Existing Nodes

```release-note
Adds a Getting Started Guide for Rancher 2.x using Existing Nodes
```

### CODE REVIEW

File: Documentation/gettingstarted/rancher.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
+****************************
+Installing Cilium on Rancher
+****************************
+
+Create a New Cluster
+=====================
```
Comment (qmonnet) [inline Documentation/gettingstarted/rancher.rst]:
```suggestion
Create a New Cluster
====================
```

File: Documentation/gettingstarted/rancher.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+In Rancher UI, navigate to the Clusters page. In the top right, click on the
+`Add Cluster` box to create a new cluster.
```
Comment (qmonnet) [inline Documentation/gettingstarted/rancher.rst]: This is RST, not markdown. The simple backquotes in RST are for “[interpreted text](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#interpreted-text)”, text interpreted in a specific context. Instead, you want inline litterals, with double-backquotes.
```suggestion
``Add Cluster`` box to create a new cluster.
```
Here and at other locations in the document. This should fix most of the complaints from the CI action.

File: Documentation/gettingstarted/rancher.rst [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+Before adding nodes to the cluster, ensure the BPF filesystem  is mounted.
+You can persist the configuration using the following commands:
+
+```
+sudo mount bpffs -t bpf /sys/fs/bpf
+sudo bash -c 'cat <<EOF >> /etc/fstab
+none /sys/fs/bpf bpf rw,relatime 0 0
+EOF'
+```
```
Comment (qmonnet) [inline Documentation/gettingstarted/rancher.rst]: This is Markdown. RST:
```suggestion
.. code-block:: shell-session

    $ sudo mount bpffs -t bpf /sys/fs/bpf
    $ sudo bash -c 'cat <<EOF >> /etc/fstab
    none /sys/fs/bpf bpf rw,relatime 0 0
    EOF'
```

Can't the second command be simplified into something like `sudo echo 'none /sys/fs/bpf bpf rw,relatime 0 0' >> /etc/fstab`?

File: Documentation/gettingstarted/rancher.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+cluster in the `Provisioning` state.
+
+.. image:: images/rancher_cluster_state_provisioning.png
+
+Rancher will also present you with instructions on how to add nodes to the
+cluster:
+
+.. image:: images/rancher_add_nodes.png
+
+Before adding nodes to the cluster, ensure the BPF filesystem  is mounted.
```
Comment (qmonnet) [inline Documentation/gettingstarted/rancher.rst]:
```suggestion
Before adding nodes to the cluster, ensure the BPF filesystem is mounted.
```

File: Documentation/gettingstarted/rancher.rst [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+```
+
+Next, add nodes to the cluster using the Rancher-provided Docker commands. Be
+sure to add the appropriate number of nodes required for your cluster. After
+a few minutes, you will see that the Nodes overview will show an error message
+in the Rancher UI:
+
+.. image:: images/rancher_node_not_ready.png
+
+This error is expected. - [ ] The Kubernetes components are not able to fully
```
Comment (qmonnet) [inline Documentation/gettingstarted/rancher.rst]: Leftover from copy-paste?
```suggestion
This error is expected. The Kubernetes components are not able to fully
```

File: Documentation/gettingstarted/rancher.rst [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+button:
+
+.. image:: images/rancher_add_catalog.png
+
+In the window that opens, add the official Cilium Helm v3 chart repository to
+the Rancher catalog:
+
+.. image:: images/rancher_add_cilium_catalog.png
+
+One added, you should see the Cilium repository in the catalog list:
```
Comment (qmonnet) [inline Documentation/gettingstarted/rancher.rst]:
```suggestion
Once added, you should see the Cilium repository in the catalog list:
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thanks Sean, the guide looks nice (although I haven't checked the pictures yet)! But the formatting is not good. You need to address it, Markdown syntax is not valid. See the suggestions inline.

In addition to the use of single backquotes, the CI complains because your document is not included anywhere in the doc, it is an orphan page not linked from any other. You need to reference it, most likely in `Documentation/gettingstarted/index.rst`.

Note that you should be able to compile and preview the documentation locally with:

    make -C Documentation live-preview

The content looks good, should we also include the instructions to run the connectivity tests, to make sure that Cilium is running as expected?

File: Documentation/gettingstarted/k8s-install-rancher-existing-nodes.rst [hunk 7]
```diff
@@ -0,0 +1,99 @@
+***********************
+Installation using Rancher
+***********************
```
Comment (qmonnet) [inline Documentation/gettingstarted/k8s-install-rancher-existing-nodes.rst]: This time, the GitHub action complains about the following:

```suggestion
**************************
Installation using Rancher
**************************
```

File: Documentation/gettingstarted/k8s-install-rancher-existing-nodes.rst [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+.. image:: images/rancher_add_nodes.png
+
+Before adding nodes to the cluster, ensure the BPF filesystem is mounted.
+You can persist the configuration using the following commands:
+
+.. code-block:: shell-session
+
+  sudo mount bpffs -t bpf /sys/fs/bpf
+  sudo echo 'none /sys/fs/bpf bpf rw,relatime 0 0' >> /etc/fstab
```
Comment (qmonnet) [inline Documentation/gettingstarted/k8s-install-rancher-existing-nodes.rst]: I'm terribly sorry, on re-reading this, I was mistaken. The redirection itself needs to be performed as root, and the `bash -c` you had was the correct thing. Apologies for that!
```suggestion
  sudo bash -c "echo 'none /sys/fs/bpf bpf rw,relatime 0 0' >> /etc/fstab"
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Two more observations, one is my fault - Sorry!

File: Documentation/gettingstarted/k8s-install-rancher-existing-nodes.rst [hunk 9]
```diff
@@ -0,0 +1,99 @@
+**************************
```
Comment (qmonnet) [inline Documentation/gettingstarted/k8s-install-rancher-existing-nodes.rst:7]: We forgot the disclaimer at the top of the file:
```suggestion
.. only:: not (epub or latex or html)

    WARNING: You are looking at unreleased Cilium documentation.
    Please use the official rendered version released here:
    https://docs.cilium.io

**************************
```

File: Documentation/gettingstarted/k8s-install-rancher-existing-nodes.rst [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+.. image:: images/rancher_add_nodes.png
+
+Before adding nodes to the cluster, ensure the BPF filesystem is mounted.
+You can persist the configuration using the following commands:
+
+.. code-block:: shell-session
+
+  sudo mount bpffs -t bpf /sys/fs/bpf
+  sudo bash -C "echo 'none /sys/fs/bpf bpf rw,relatime 0 0' >> /etc/fstab"
```
Comment (qmonnet) [inline Documentation/gettingstarted/k8s-install-rancher-existing-nodes.rst]: This should be a lower case `-c`.

```suggestion
  sudo bash -c "echo 'none /sys/fs/bpf bpf rw,relatime 0 0' >> /etc/fstab"
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
You're welcome!
A last few comments. While looking at other guides I realised we're missing the disclaimer at the top of the file, see my comment below. There's also a typo on the bash option (should be lowercase `-c`, see below).

After that, it should be all from my side. I built the doc locally and tested that everything works as expected. The doc build fine and is linked from the list of GSG, and the links to the pictures are correct and consistent with the walk-through, all good on that side :)

**Review by qmonnet — APPROVED** [review]
Looks good to me now, thanks a lot! Could you please squash your two commits into a single one?

**Comment by seanmwinn** [comment]
@qmonnet all squashed and ready, thanks again

**Pull request merged**